### PR TITLE
fix: XHR req method capture

### PR DESCRIPTION
**Pull request by pauldambra**
We were using `req.method` _before_ the request was used and so XHR was always captured as "GET"

### CODE REVIEW

File: cypress/e2e/session-recording.cy.ts [hunk 1]
```diff
@@ -280,6 +280,62 @@ describe('Session recording', () => {
                     )
                 })
             })
+
+            it('it captures XHR method correctly', () => {
```
Comment (pauldambra) [inline cypress/e2e/session-recording.cy.ts:284]: these tests have a lot of duplication now, but I want to get this fix out

File: src/entrypoints/recorder.ts [hunk 2]
```diff
@@ -386,7 +386,7 @@ function prepareRequest({
             timeOrigin,
             timestamp,
             method: method,
-            initiatorType: entry ? (entry.initiatorType as InitiatorType) : initiatorType,
```
Comment (pauldambra) [inline src/entrypoints/recorder.ts:389]: flips the choice here... if we're providing an initiator we should use it in preference to the detected one

they _should_ always match

File: src/extensions/replay/sessionrecording.ts [hunk 3]
```diff
@@ -430,10 +430,8 @@ export class SessionRecording {
     }
 
     /**
-     * Any one trigger can activate the session
-     * So, if they are all the same - return that value
-     * If either is disabled return the other's valu
-     * @private
+     * trigger is active if _either_ URL _or_ event trigger is active
+     * and is pending if _either_ URL _or_ event trigger is pending
```
Comment (pauldambra) [inline src/extensions/replay/sessionrecording.ts:434]: flyby realised this comment was nonsense 🤣

File: src/entrypoints/recorder.ts [hunk 4]
```diff
@@ -309,7 +309,7 @@ function initXhrObserver(cb: networkCallback, win: IWindow, options: Required<Ne
                         .then((entry) => {
                             const requests = prepareRequest({
                                 entry,
-                                method: req.method,
+                                method: method,
```
Comment (pauldambra) [inline src/entrypoints/recorder.ts:312]: this is the fix... the method is provided to the XHR function when it is called so we should use it

File: src/extensions/replay/sessionrecording.ts (hunk 3 above)
Comment (robbie-c) [reply src/extensions/replay/sessionrecording.ts:434]: Pedantic brain: what if one is active and one is pending?

File: cypress/e2e/session-recording.cy.ts (hunk 1 above)
Comment (robbie-c) [reply cypress/e2e/session-recording.cy.ts:284]: This test is pretty convincing, how would it have failed before?

Comment (robbie-c) [reply cypress/e2e/session-recording.cy.ts:284]: The method = POST assertion?

**Review by robbie-c — APPROVED** [review]
We're deep within code I haven't looked at before so with that caveat:
* The bug itself makes sense
* The fix for it makes sense
* The test looks good

👍 👍 👍

File: cypress/e2e/session-recording.cy.ts (hunk 1 above)
Comment (pauldambra) [reply cypress/e2e/session-recording.cy.ts:284]: yep... it was always GET before (or at least overwhelmingly GET I didn't check which)

File: src/extensions/replay/sessionrecording.ts (hunk 3 above)
Comment (pauldambra) [reply src/extensions/replay/sessionrecording.ts:434]:
```suggestion
```

also... i'm describing the code... it's a redundant comment

File: src/extensions/replay/sessionrecording.ts [hunk 5]
```diff
@@ -430,10 +430,6 @@ export class SessionRecording {
     }
 
     /**
-     * Any one trigger can activate the session
-     * So, if they are all the same - return that value
-     * If either is disabled return the other's valu
-     * @private
      */
```
Comment (pauldambra) [inline src/extensions/replay/sessionrecording.ts:433]:
```suggestion
```

File: src/extensions/replay/sessionrecording.ts [hunk 6]
```diff
@@ -430,11 +430,6 @@ export class SessionRecording {
     }
 
     /**
```
Comment (pauldambra) [inline src/extensions/replay/sessionrecording.ts:432]:
```suggestion
```

File: cypress/e2e/session-recording.cy.ts (hunk 1 above)
Comment (pauldambra) [reply cypress/e2e/session-recording.cy.ts:284]: and for clarity I validated the test failed before the change 😊

**Pull request merged**